### PR TITLE
Messages selection

### PR DESCRIPTION
**Pull request by AntonPalich**
* Added support for selection indicator to `BaseMessagesCollectionViewCell` 
* Implemented "Messages Selection" feature in the example app that uses selection indicator from `BaseMessagesCollectionViewCell` to select/deselect messages in the chat

### CODE REVIEW

File: ChattoAdditions/Source/Chat Items/BaseMessage/BaseMessagePresenter.swift [hunk 1]
```diff
@@ -101,6 +103,8 @@ open class BaseMessagePresenter<BubbleViewT, ViewModelBuilderT, InteractionHandl
         cell.performBatchUpdates({ () -> Void in
             self.messageViewModel.showsTail = decorationAttributes.canShowTail
             self.messageViewModel.showsAvatar = decorationAttributes.canShowAvatar
+            self.messageViewModel.isShowingSelectionIndicator = decorationAttributes.isShowingSelectionIndicator
```
Comment (geegaset) [inline ChattoAdditions/Source/Chat Items/BaseMessage/BaseMessagePresenter.swift]: the way how you copy all the attributes from one entity to another looks a little bit suspicious. why don't you have the dedicated entity which represents the whole set of properties? in this way copying would look like: self.mesageViewModel.attributes = decorationAttributes.attributes

File: ChattoAdditions/Source/Chat Items/BaseMessage/Views/BaseMessageCollectionViewCell.swift [hunk 2]
```diff
@@ -325,7 +336,28 @@ open class BaseMessageCollectionViewCell<BubbleViewType>: UICollectionViewCell,
...
 
+    // MARK: Selection
+
+    private let selectionIndicator = UIImageView(frame: .zero)
+
+    private func updateSelectionIndicator(with style: BaseMessageCollectionViewCellStyleProtocol) {
+        self.selectionIndicator.image = style.selectionIndicatorIcon(for: self.messageViewModel)
+    }
+
+    private lazy var selectionTapGestureRecognizer: UITapGestureRecognizer = {
```
Comment (geegaset) [inline ChattoAdditions/Source/Chat Items/BaseMessage/Views/BaseMessageCollectionViewCell.swift]: does it really need to be lazy? it looks like it's used in init.

File: ChattoAdditions/Source/Chat Items/BaseMessage/Views/BaseMessageCollectionViewCell.swift [hunk 3]
```diff
@@ -364,34 +396,65 @@ fileprivate struct Layout {
...
+        let selectionIndicatorSize = parameters.selectionIndicatorSize
 
         let preferredWidthForBubble = (containerWidth * parameters.maxContainerWidthPercentageForBubbleView).bma_round()
         let bubbleSize = bubbleView.sizeThatFits(CGSize(width: preferredWidthForBubble, height: .greatestFiniteMagnitude))
         let containerRect = CGRect(origin: CGPoint.zero, size: CGSize(width: containerWidth, height: bubbleSize.height))
 
-        self.bubbleViewFrame = bubbleSize.bma_rect(inContainer: containerRect, xAlignament: .center, yAlignment: .center)
-        self.failedButtonFrame = failedButtonSize.bma_rect(inContainer: containerRect, xAlignament: .center, yAlignment: .center)
-        self.avatarViewFrame = avatarSize.bma_rect(inContainer: containerRect, xAlignament: .center, yAlignment: parameters.avatarVerticalAlignment)
+        self.bubbleViewFrame = bubbleSize.bma_rect(
```
Comment (geegaset) [inline ChattoAdditions/Source/Chat Items/BaseMessage/Views/BaseMessageCollectionViewCell.swift]: is this correct to use bma_ prefixes in Chatto code?

Comment (AntonPalich) [reply ChattoAdditions/Source/Chat Items/BaseMessage/Views/BaseMessageCollectionViewCell.swift]: https://github.com/badoo/Chatto/issues/410

File: ChattoAdditions/Source/Chat Items/BaseMessage/Views/BaseMessageCollectionViewCell.swift (hunk 2 above)
Comment (AntonPalich) [reply ChattoAdditions/Source/Chat Items/BaseMessage/Views/BaseMessageCollectionViewCell.swift]: I'll fix it

**Pull request merged**